tests/migrations/test_runner.py

```diff
@@ -209,6 +209,56 @@ def test_reverse_all() -> None:
     ), "All tables should be deleted"
 
 
+def test_reverse_idempotency_all() -> None:
+    # This test is to ensure that reversing a migration twice does not cause any
+    # issues or unintended side effects. This is important because we may need to reverse
+    # a migration multiple times in the event of a rollback.
+    runner = Runner()
+    all_migrations = runner._get_pending_migrations()
+    runner.run_all(force=True)
+    connection = get_cluster(StorageSetKey.MIGRATIONS).get_query_connection(
+        ClickhouseClientSettings.MIGRATE
+    )
+    for migration in reversed(all_migrations):
+        runner.reverse_migration(migration, force=True)
+        if migration.group != MigrationGroup.SYSTEM:
+
+            def reverse_twice() -> None:
+                # reverse again to ensure idempotency
+                runner.run_migration(migration, fake=True)
+                runner.reverse_migration(migration, force=True)
+                runner.run_migration(migration, fake=True)
+                runner.reverse_migration(migration, force=True)
+
+            group = migration.group.value
+
+            # for some reason this has to be here. it looks like
```

Comment:
It would  be good to understand what's happening here.

Reply:
Guessing it's because some other tests are doing stuff like this: https://github.com/getsentry/snuba/blob/3cff084ed27b6cb013437f8e9cb1add1f778d847/tests/clusters/test_cluster.py#L107